core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -149,6 +160,33 @@ private ManifestFiles() {
     throw new UnsupportedOperationException("Cannot write manifest for table version: " + formatVersion);
   }
 
+  private static org.apache.avro.Schema getManifestAvroSchema() {
```

Comment:
I would rather not leak an Avro class from this API. Can we put these methods in a utility in the Avro package?

Reply:
Yeah,  in theory we should not expose any avro classes, GenericAvroReader, GenericAvroWriter (or any other internal classes ) to public API. It's a draft patch before,  so I tried to implement this as soon as possible and ignored those points.  Let me make it more reasonable.

Reply:
We had discussed this before and I also submitted #1219 that start to use Iceberg model instead of Avro's `IndexedRecord` for `GenericManifestFile`.

Reply:
Skimmed that pr,  looks like we are changing the GenericManifestFile from an IndexRecord to an Iceberg Record, and use the `DataReader` and `DataWriter` to accomplish IO.   What's the purpose that we turn to Iceberg record ?  For unifying the `IndexRecord` and `Record` and keeping the `Record`  only in future ?

Reply:
Yes, the purpose is to make internal classes not depend on Avro data model. After we replace the Avro data mode with Iceberg data model, we could refactor two Avro readers and writers as mentioned in https://github.com/apache/iceberg/issues/1152.

Reply:
OK, thanks for the reminding,  it's worth to finish that.